src/value/owned.rs

```diff
@@ -0,0 +1,228 @@
+// Copyright Amazon.com, Inc. or its affiliates.
+
+//! Provides owned implementations of [`SymbolToken`], [`Element`] and its dependents.
+//!
+//! This API is simpler to manage with respect to borrowing lifetimes, but requires full
+//! ownership of data to do so.
+
+use super::{AnyInt, Element, ImportSource, Sequence, Struct, SymbolToken};
+use crate::types::SymbolId;
+use crate::IonType;
+
+/// An owned implementation of  [`ImportSource`].
+#[derive(Debug, Clone)]
+pub struct OwnedImportSource {
+    table: String,
```

Comment:
Given that any symbol table import is going to end up creating lots of references to the same table name (one per `SymbolToken`), I think that the `table` field is probably going to be an `Rc<String>`. That doesn't need to hold up this PR though, since `table` isn't part of the trait.

Reply:
Added a TODO for this `OwnedSymbolTable::text`.